### ISSUE TITLE
mws-simple does not work with /Products/GetLowestPricedOffersForASIN

### ISSUE THREAD

**Issue by ericblade** [ericblade/mws-advanced#17]
As per this thread: https://stackoverflow.com/questions/36293752/getlowestpricedoffersforsku-failed-processing-arguments

I was just  attempting to implement this call, and sure enough, I get the exact same problem, an error 400, followed by 

    Failed processing arguments of org.jboss.resteasy.spi.metadata.ResourceMethod@1f3d6bd9

There are a few things needed to be done here:

 1- Trapping the error 400 -- I had to go all the way back to mws-simple, just to get the dump of the data. So, no data other than a blank Array is coming back to mws-advanced.  I'm not sure if ultimately, this means forking and patching mws-simple, which hasn't been updated in a year, monkey patching it, or just absorbing it so that we can add proper functionality to it for the above call, per th e stack overflow answer.  In any case, we need to capture the error, and treat it AS an error.  Checking response 'body.statusCode < 200 || body.statusCode > 299' should give us some valid reasonable error checking, right around line 96 or so of mws-simple.  Then we should also treat that as an error going back to line 99-ish in request.post() . . at which point then the clients will know that something has seriously broken.

2- actually writing a new method to handle this.  We need to review the above link to Stack Overflow, and then determine if that is a thing that we can add to *every* call, and it will make them all "Just Work"(tm), or if this call method only works with the functions that the existing one does not (the one referenced here, and the additional one(s) referenced in the Stack thread.

3- Before we fix this, we should make sure that we have a way of replicating it, so that we can write a test that checks that everyone handles exceptions like they should. Like, by calling this specific endpoint in raw mode, and checking it's returns, etc.  If we can generate a 400 error from Amazon's end, then we should be able to write a test that can show it, so that maybe they can fix it, or something like that.

**Comment by ericblade** [comment]
Worthy of noting, that I have created a fork of mws-simple, specifically for the purpose of fixing this problem.  Since I created it, I also removed the monkeypatch that was in mws-advanced back to it, for using MWSAuthToken .. and I cherry-picked some fixes from other forks.  It doesn't look like anyone's paying too much attention to that lib.. so maybe just time to overhaul it completely, and make sure it works for all types of requests.
